protocol/grpc_new/client.go

```diff
@@ -0,0 +1,167 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package grpc_new
+
+import (
+	"context"
+	"crypto/tls"
+	connect "dubbo.apache.org/dubbo-go/v3/protocol/grpc_new/connect"
+	"golang.org/x/net/http2"
+	"net"
+	"net/http"
+	"reflect"
+	"strings"
+)
+
+import (
+	"github.com/dubbogo/gost/log/logger"
+
+	"github.com/dustin/go-humanize"
+)
+
+import (
+	"dubbo.apache.org/dubbo-go/v3/common"
+	"dubbo.apache.org/dubbo-go/v3/common/constant"
+	"dubbo.apache.org/dubbo-go/v3/config"
+)
+
+const (
+	httpPrefix  string = "http://"
+	httpsPrefix string = "https://"
+)
+
+// Client is GRPC_NEW client wrapping connection and dubbo invoker
+type Client struct {
+	*http.Client
+	invoker reflect.Value
+}
+
+// NewClient creates a new GRPC_NEW client.
+func NewClient(url *common.URL) (*Client, error) {
+	// If global trace instance was set, it means trace function enabled.
+	// If not, will return NoopTracer.
+	//tracer := opentracing.GlobalTracer()
```

Comment:
```
// If global trace instance was set, it means trace function enabled.
// If not, it will return NoopTracer.
// tracer := opentracing.GlobalTracer()
```

Reply:
I need to think about how to use Middleware mode to add tracer.